nums/numpy/api.py

```diff
@@ -444,6 +446,29 @@ def any(a: BlockArray, axis=None, out=None, keepdims=False):
     return _instance().reduce("any", a, axis=axis, keepdims=keepdims)
 
 
+def average(a: BlockArray,
+            axis: Union[None, int] = None,
+            weights: Union[None, BlockArray] = None,
+            returned: bool = False):
+    if weights is None:
+        avg = mean(a, axis)
+        weights_sum = BlockArray.from_scalar(a.size / avg.size, a.system)
```

Comment:
Any clean way to compute `weights_sum` here only if it's returned? It's very low overhead. It's okay to keep as-is to maintain readability.